src/temporal_logic/rules.rs

```diff
@@ -1301,6 +1301,13 @@ pub proof fn wf1_assume<T>(spec: TempPred<T>, next: ActionPred<T>, forward: Acti
     wf1_variant_assume_temp::<T>(spec, lift_action(next), lift_action(forward), lift_state(asm), lift_state(p), lift_state(q));
 }
 
+/// Leads to p and q if (1) p leads to q and (2) ~q and next preserve p
+/// pre:
+///     spec |= [](p /\ ~q /\ next => p')
+///     spec |= p ~> q
+///     spec |= []next
+/// post:
+///     spec |= p ~> p /\ q
 pub proof fn leads_to_confluence_self_temp<T>(spec: TempPred<T>, next: TempPred<T>, p: TempPred<T>, q: TempPred<T>)
```

Comment:
I think the proof of this lemma can be simplified using `leads_to_partial_confluence_temp`. I can take a try.

Reply:
If so, you can also delete this lemma and use the one you mention to replace it in the proof.